bucket.go

```diff
@@ -130,7 +131,7 @@ type UpdateFunc func(current []byte) (updated []byte, err error)
 
 type WriteUpdateFunc func(current []byte) (updated []byte, opt WriteOptions, err error)
 
-type WriteUpdateWithXattrFunc func(doc []byte, xattr []byte, cas uint64) (updatedDoc []byte, updatedXattr []byte, err error)
+type WriteUpdateWithXattrFunc func(doc []byte, xattr []byte, cas uint64) (updatedDoc []byte, updatedXattr []byte, deletedDoc bool, err error)
```

Comment:
Can you add a parameter description for `deletedDoc`.  It looks like a good variable name, but I think some expanded description might help .. like "Perform the update on deleted docs if deletedDoc is true, but return an error if an updated is attempted to be performed on a deleted doc"  <-- example only, no idea if that's the behavior!